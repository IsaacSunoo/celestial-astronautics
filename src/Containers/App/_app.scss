* {
  box-sizing: border-box;
  margin: 0;
}

.app {
    text-align: center;
    img {
        height: 400px;
        width: auto;
    }
}