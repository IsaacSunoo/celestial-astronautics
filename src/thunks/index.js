import { setLoading, getApolloData, setError, getDaily } from '../actions';
import { fetchCall } from '../api';
import apiKey from '../api_key/apiKey';

// const baseUrl = 'https://api.nasa.gov/';

export const getApollo20 = () => {
    const url = 'https://images-api.nasa.gov/search?q=apollo%20landing&media_type=image';
    return async dispatch => {
        try {
            dispatch(setLoading(true));
            const apollo = await fetchCall(url);
            dispatch(setLoading(false));
            dispatch(getApolloData(apollo));
        } catch (err) {
            dispatch(setError(err.message));
        }
    }
}

export const getDailyNews = () => {
    const url = `https://api.nasa.gov/planetary/apod?api_key=${apiKey}`;
    return async dispatch => {
        try {
            dispatch(setLoading(true));
            const dailyNews = await fetchCall(url);
            dispatch(setLoading(false));
            dispatch(getDaily(dailyNews));
        } catch (err) {
            dispatch(setError(err.message));
        }
    }
}

