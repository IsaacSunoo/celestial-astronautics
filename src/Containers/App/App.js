import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import { connect } from 'react-redux';
import { setLoading } from '../../actions';
import { getDailyNews } from '../../thunks';
import LoadingGif from '../../images/Loading.gif';
import PageNotFound from '../../components/PageNotFound';
import Header from '../../components/Header';
import Apollo20 from '../Apollo20';
import Orion from '../Orion';
import MainPage from '../../components/MainPage';
import Milkyway from '../Milkyway';
import Scratchpad from '../Scratchpad';
import Nebula from '../Nebula';
import Galaxy from '../Galaxy';
import Stars from '../Stars';
// import PropTypes from 'prop-types';

class App extends Component {
  componentDidMount = async () => {
    const { getDailyNews } = this.props;
    getDailyNews();
  }

  render() {
    const { dailyNews, isLoading } = this.props;

    return (
      <div className='app'>
        <Route path='/' component={Header} />
          <Switch>
            <Route exact path='/' render={() => (
              <MainPage {...dailyNews}/>
            )} />
            <Route exact path='/apollo' component={Apollo20} />
            <Route exact path='/orion' component={Orion} />
            <Route exact path='/milkyway' component={Milkyway} />
            <Route exact path='/nebula' component={Nebula} />
            <Route exact path='/galaxy' component={Galaxy} />
            <Route exact path='/stars' component={Stars} />
            <Route exact path='/scratchpad' component={Scratchpad} />
            <Route component={PageNotFound} />
          </Switch>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  isLoading: state.isLoading,
  dailyNews: state.dailyNews
});

export const mapDispatchToProps = dispatch => ({
  setLoading: (bool) => dispatch(setLoading(bool)),
  getDailyNews: () => dispatch(getDailyNews())
});

// App.propTypes = {
//   dailyNews: PropTypes.array.isRequired,
//   getDailyNews: PropTypes.func.isRequired
// }

export default connect(mapStateToProps, mapDispatchToProps)(App);
