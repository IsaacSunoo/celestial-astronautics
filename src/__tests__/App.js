import React from 'react';
import { App, mapDispatchToProps, mapStateToProps } from '../containers/App/App';
import { shallow } from 'enzyme';
import { getDailyNews } from '../thunks';

jest.mock('../thunks');

describe('App', () => {
    let wrapper;
    window.fetch = jest.fn();
    beforeEach(() => {
        wrapper = shallow(<App />);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

      describe('mapStateToProps', () => {
        it('returns an object with dailyNews', () => {
          const mockState = {
            dailyNews: ['Daily', 'news'],
            error: 'error'
          };
          const expectedState = {
            dailyNews: ['Daily', 'news']
          };
          const mappedProps = mapStateToProps(mockState);
          expect(mappedProps).toEqual(expectedState);
        });
      });

      describe('mapDispatchToProps', () => {
        it('calls dispatch with a getGalaxy thunk', () => {
          const mockDispatch = jest.fn();
          const actionToDispatch = getDailyNews();
          const mappedProps = mapDispatchToProps(mockDispatch);
          mappedProps.getDailyNews();
          expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
        });
      });
});