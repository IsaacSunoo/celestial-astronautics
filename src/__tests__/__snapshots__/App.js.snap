// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="app"
>
  <Route
    component={[Function]}
    path="/"
  />
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Apollo20)",
          "type": [Function],
          Symbol(Symbol.hasInstance): [Function],
        }
      }
      exact={true}
      path="/apollo"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Orion)",
          "type": [Function],
          Symbol(Symbol.hasInstance): [Function],
        }
      }
      exact={true}
      path="/orion"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Milkyway)",
          "type": [Function],
          Symbol(Symbol.hasInstance): [Function],
        }
      }
      exact={true}
      path="/milkyway"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Nebula)",
          "type": [Function],
          Symbol(Symbol.hasInstance): [Function],
        }
      }
      exact={true}
      path="/nebula"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Galaxy)",
          "type": [Function],
          Symbol(Symbol.hasInstance): [Function],
        }
      }
      exact={true}
      path="/galaxy"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Stars)",
          "type": [Function],
          Symbol(Symbol.hasInstance): [Function],
        }
      }
      exact={true}
      path="/stars"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;
